src/components/SQForm/SQFormIconButton.tsx

```diff
@@ -14,7 +14,7 @@ interface SQFormIconButtonProps {
   /** Type of button, defaults to 'submit' */
   type?: string;
   /** Standard React event handler */
-  onClick?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onClick?: (...event: React.MouseEvent<HTMLButtonElement>[]) => void;
```

Comment:
Sorry, I should've been more clear in my other comment - we want the MouseEvent type that you had before, but not the event spread and `[]`

```suggestion
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
```